### ISSUE TITLE
forwardDebugPort does not work correctly if more than one device / emulator

### ISSUE THREAD

**Issue by cancan101** [realm/realm-js#356]
```
:realm:forwardDebugPort
error: more than one device/emulator
===========================================================================
WARNING: Failed to automatically forward port 8082.
In order to use Realm in Chrome debugging mode, port 8082 must be forwarded
from localhost to the device or emulator being used to run the application.
You may need to add the appropriate flags to the command that failed:
    adb forward tcp:8082 tcp:8082
===========================================================================
```

**Comment by appden** [comment]
@cancan101 this is somewhat expected because when using Chrome debug mode, the JS that's running in Chrome needs to make HTTP requests to `localhost:8082`, which should forward to that port on the device/emulator running the app. So if multiple devices are connected, then you would need to manually tell `adb forward` which device to forward that port to. See the `-d`, `-e`, and `-s` options in `adb -h`. You can list them with `adb devices -l`.

**Comment by cancan101** [comment]
Yes, I know the command on the CLI. Is there no way to get the correct value from the launch configs and set it here?

**Comment by appden** [comment]
I'm not really sure if there is a way to do that here. If you have any ideas, I'd love to hear them!

It's definitely something we need to address though considering they changed `react-native run-android` to run on all connected devices (see [this commit](https://github.com/facebook/react-native/commit/10ad47a711954f1af077c7f237ad37ff269d324e)).

**Comment by cancan101** [comment]
What exactly is this port forwarding for anyways? At first I thought it might be realm browser, but I don't think that is the case?

**Comment by alazier** [comment]
@cancan101 port forwarding is for chrome debug mode.

**Comment by cancan101** [comment]
What is special about realm here that it needs to set up this forwarding?

**Comment by alazier** [comment]
Unlike other react-native modules Realm apis are synchronous and are built directly on top of the JS engine. We do this for good performance (we don't have to copy data for every call) and api simplicity. This is why you are able to access object properties using normal property getters/setters rather than using async callbacks which we thought would be cumbersome and awkward.

To make chrome debugging work with this design we had to build an rpc server that allows us to synchronously forward all api calls so they are run on the device/emulator. This rpc server is what requires port forwarding and is only used when debugging in chrome.

**Comment by alazier** [comment]
We should mention this in the docs.
